src/lib/dhcp/option6_auth.h

```diff
@@ -0,0 +1,143 @@
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef OPTION6_AUTH_H
+#define OPTION6_AUTH_H
+#endif
+
+#include <dhcp/option.h>
+#include <boost/shared_ptr.hpp>
+
+#include <vector> 
+
+namespace isc {
+namespace dhcp {
+
+class Option6Auth;
+
+/// A pointer to the @c isc::dhcp::Option6Auth object.
+typedef boost::shared_ptr<Option6Auth> Option6AuthPtr;
+
+class Option6Auth: public Option {
+
+public:
+    
+    static const uint8_t OPTION6_AUTH_MIN_LEN  = 11;
+    static const uint8_t OPTION6_HASH_MSG_LEN  = 16;
+    static const uint8_t OPTION6_HDR = 4; 
+    /// @brief Constructor, used for auth options while transmitting
+    ///
+    /// @param proto protocol type
+    /// @param algo algorithm type
+    /// @param method remote detection method
+    /// @param rdm replay detection value
+    /// @param info authentication info.
+    Option6Auth(const uint8_t proto, const uint8_t algo, const uint8_t method,
+                const uint64_t rdm, const std::vector<uint8_t>& info);
+
+    /// @brief Copies this option and returns a pointer to the copy.
+    virtual OptionPtr clone() const;
+
+    /// Writes option in wire-format to buf, returns pointer to first unused
+    /// byte after stored option.
+    ///
+    /// @param buf buffer (option will be stored here)
+    void pack(isc::util::OutputBuffer& buf) const;
+
+    /// Writes option in wire-format to buf, for computing hash  
+    /// auth info filled with 0 for a length of 128 bits
+    /// returns with pointer to first unused
+    /// byte after stored option.
+    ///
+    /// @param buf buffer (option will be stored here)
+    void packHashInput(isc::util::OutputBuffer& buf) const;
+
+    /// @brief Parses received buffer
+    /// @brief Parses received buffer
+    ///
+    /// Parses received buffer and returns offset to the first unused byte after
+    /// parsed option.
+    ///
+    /// @param begin iterator to first byte of option data
+    /// @param end iterator to end of option data (first byte after option end)
+    virtual void unpack(OptionBufferConstIter begin, OptionBufferConstIter end);
+
+    /// Provides human readable text representation
+    ///
+    /// @param indent number of leading space characters
+    ///
+    /// @return string with text representation
+    virtual std::string toText(int indent = 0) const;
+
+    /// Set protocol type
+    ///
+    /// @param proto protocol type to be set
+    void setProtocol(uint8_t proto) { protocol_ = proto; }
+
+    /// Set hash alogrithm type
+    ///
+    /// @param algo hash alogrithm type to be set
+    void setHashAlgo(uint8_t algo) { algorithm_ = algo; }
+
+    /// Set replay detection method type
+    ///
+    /// @param method replay detection method to be set
+    void setRplyDtctnMthd(uint8_t method) { rdm_method_ = method; }
```

Comment:
Please avoid abbreviation like this. The problem here is that you won't be able to remember them without looking at the header file.